Lib/sqlite3/test/regression.py

```diff
@@ -376,6 +377,20 @@ def CheckCommitCursorReset(self):
                 counter += 1
         self.assertEqual(counter, 3, "should have returned exactly three rows")
 
+    def CheckBpo31770(self):
+        """
+        The interpreter shouldn't crash in case Cursor.__init__() is called
+        more than once.
+        """
+        def callback(*args):
+            pass
+        con = sqlite.connect(":memory:")
+        cur = sqlite.Cursor(con)
+        ref = weakref.ref(cur, callback)
+        cur.__init__(con)
+        del cur
+        del ref  # shouldn't crash
```

Comment:
I don't see the purpose of removing explicitly the last reference to a weak reference object. Just remove "del ref" ?

Reply:
At least on my Windows 10, when the test fails, the `del ref` causes the crash info that is printed to include the line (and test) that caused the crash
(i.e. `File "C:\Users\orenm\cpython\lib\sqlite3\test\regression.py", line 392 in CheckBpo31770`). If i remove the `del ref`, it doesn't show info about the line and test that caused the crash.
If you still think that i should remove it, i wouldn't mind, of course.

Reply:
I would add `test.support.gc_collect()` for ensuring that objects are collected.

Reply:
When i use `test.support.gc_collect()`, the crash info also doesn't include the line and the test that crashed.
Also, when i add `1/0` after the call to `gc_collect()`, and run the test (on cpython without the patch in C code), the code doesn't crash in `gc_collect()`, and a `ZeroDivisionError` is raised...
(of course, without the `1/0`, it does crash, but after the `gc_collect()`, i guess.)

Reply:
Should i replace the `del ref` with a call to `test.support.gc_collect()` anyway?

Reply:
Did you call `test.support.gc_collect()` *instead* of `del ref`? Or adding `test.support.gc_collect()` *after* `del ref` breaks the test?

I meant that you should keep `del ref`, but call `test.support.gc_collect()` after it for sure. Of course, if this breaks the test, don't add it.